src/client/api/event.c

```diff
@@ -799,43 +794,32 @@ daos_eq_query(daos_handle_t eqh, daos_eq_query_t query,
 static void
 daos_event_abort_one(struct daos_event_private *evx)
 {
-	if (evx->evx_status != DAOS_EVS_RUNNING)
-		return;
-
 	/* NB: ev::ev_error will be set by daos_event_complete(),
 	 * so user can decide to not set error if operation has already
 	 * finished while trying to abort */
-	/* NB: always set ev_status to DAOS_EVS_ABORTED even w/o callback,
+	/* NB: always set ev_status to DAOS_EVS_ABORTING even w/o callback,
 	 * so aborted parent event can be marked as COMPLETE right after
 	 * completion all launched events other than completion of all
 	 * children. See daos_parent_event_can_complete for details. */
-	evx->evx_status = DAOS_EVS_ABORTED;
+	evx->evx_status = DAOS_EVS_ABORTING;
 	daos_event_complete_cb(evx, -DER_CANCELED);
```

Comment:
here set the status as ABORTING, but call event's complete cb, it will tell the user it is completed and can free/reuse the event? 
But actually you want to ask user to wait it's real completion by eq_poll by email discussion?

Reply:
no it does not free the event, it just calls the user completion callback with the cancelled RC. that does not free buffers. in DAOS events we don't really use the event completion cb.
the user will still need to call poll for the event completion.

Reply:
ok, daos_event_register_comp_cb() not used by any daos code now, but from API comments when it get called it can mean the event completed, then user possibly free/reuse it ...
may confirm with @gnailzenh 
otherwise the patch looks fine to me.

Reply:
Probably the simplest workaround is still just change abort() and make it be NOOP (there is nothing we can actually do at this point)

Reply:
OK i refreshed the patch to make abort a no-op for now.